### PR TITLE
Fixing #2850 - Part 1

### PR DESCRIPTION
**Pull request by soerenbernstein**
Preparation for overriding the filter creation

Created new protected method applyFilters(), which can be overridden

### CODE REVIEW

File: Datagrid/Datagrid.php [hunk 1]
```diff
@@ -281,4 +276,16 @@ public function getForm()
 
         return $this->form;
     }
+
+    protected function applyFilters()
```
Comment (core23) [inline Datagrid/Datagrid.php]: Please add phpdoc

**Comment by core23** [comment]
Don't know why you should override the applyFilters method? Can you give a concreate case, when you need this @soerenbernstein ?

**Comment by greg0ire** [comment]
Also, can you squash your commits together and follow these rules : http://rakeroutes.com/blog/deliberate-git/ ?

**Comment by soerenbernstein** [comment]
@core23 Have a look at #3160: Wrong search results when using custom query (Followup) for any information
@greg0ire Sorry, but no. These PRs are over a year old now and I have other things to do right now. Back then I did what was suggested to fix my first PR on this matter, which what a rather dirty hack - or let's say proof of concept. These patches might not ever apply properly anymore. So, take it or leave it - I don't care anymore.

**Comment by Soullivaneuh** [comment]
@soerenbernstein I understand your point of view. In this case I'll close this one.

If anybody want to put it back and finish the job, feel free to open another one.

**Pull request closed without merging**